diff -ru collectd-5.0.0.orig/src/irq.c collectd-5.0.0/src/irq.c
--- collectd-5.0.0.orig/src/irq.c	2011-03-28 01:35:08.000000000 +0900
+++ collectd-5.0.0/src/irq.c	2011-08-25 05:01:39.552971146 +0900
@@ -182,6 +182,11 @@
 			value_t v;
 			int status;
 
+                        if( (strcmp("PCI-MSI-edge",fields[i]) == 0) ||
+			    (strcmp("IO-APIC-fasteoi",fields[i]) == 0) ||
+			    (strcmp("IO-APIC-edge",fields[i]) == 0) )
+				break;
+
 			status = parse_value (fields[i], &v, DS_TYPE_DERIVE);
 			if (status != 0)
 				break;
